src/main/java/com/google/devtools/build/lib/skyframe/CompletionFunction.java

```diff
@@ -560,24 +447,43 @@ private Reset informImportantOutputHandler(
                     ? importantArtifacts
                     : Iterables.filter(importantArtifacts, artifact -> !artifact.isFileset()),
                 ctx,
-                metadataProvider);
+                metadataProvider,
+                artifact ->
+                    checkNotNull(
+                        ActionUtils.getActionForLookupData(env, artifact.getGeneratingActionKey()),
+                        "generating action for artifact %s",
+                        artifact));
       }
       if (lostOutputs.isEmpty()) {
         return null;
       }
 
+      Iterable<Artifact> artifactsRelevantForRewinding = importantArtifacts;
+      // Runfiles are not considered important outputs, but can arise as dep keys to which lost
+      // outputs are attributed in Bazel (but not Blaze).
+      var hiddenTopLevelArtifacts =
+          artifactsToBuild.getAllArtifactsByOutputGroup().get(OutputGroupInfo.HIDDEN_TOP_LEVEL);
+      if (hiddenTopLevelArtifacts != null) {
+        artifactsRelevantForRewinding =
+            Iterables.concat(
+                artifactsRelevantForRewinding, hiddenTopLevelArtifacts.getArtifacts().toList());
+      }
+
       // Filter out lost outputs from the set of built artifacts so that they are not reported. If
       // rewinding is successful, we'll report them later on.
       for (ActionInput lostOutput : lostOutputs.byDigest().values()) {
         builtArtifacts.remove(lostOutput);
-        builtArtifacts.removeAll(lostOutputs.owners().getOwners(lostOutput));
+        lostOutputs
```

Comment:
@justinhorvitz Is it okay that Bazel skips this part because it doesn't populate owners? I would think that the answer is no, which would probably require moving this computation into `ActionRewindStrategy.prepareRewindPlanForLostTopLevelOutputs`?

Reply:
I added a commit with this change for now.

Reply:
The removal from `builtArtifacts` is relevant for errorful cases where `builtArtifacts` is passed to `postFailedEvent`. I believe that the `_partiallyBuiltTarget_` rewinding test cases are supposed to cover this. The owners would be covered by the tree artifact one.